modules/gapi/src/backends/cpu/serialization.cpp

```diff
@@ -0,0 +1,471 @@
+                break;
+            case detail::OpaqueKind::CV_SIZE:
+                std::cout << "putOp    cv::Size " << op.args[i].get<cv::Size>().width << "x"
+                          << op.args[i].get<cv::Size>().height << std::endl;
+                sop.opaque_cvsizes.push_back(op.args[i].get<cv::Size>());
+                break;
+            case detail::OpaqueKind::BOOL:
+                std::cout << "putOp    bool " << op.args[i].get<bool>() << std::endl;
+                sop.opaque_bools.push_back(op.args[i].get<bool>());
+                break;
+            case detail::OpaqueKind::CV_SCALAR:
+                std::cout << "putOp    cv::Scalar " << op.args[i].get<cv::Scalar>()[0] << " "
+                          << op.args[i].get<cv::Scalar>()[1] << " "
+                          << op.args[i].get<cv::Scalar>()[2] << " "
+                          << op.args[i].get<cv::Scalar>()[3] << " "
+                          << std::endl;
+                sop.opaque_cvscalars.push_back(op.args[i].get<cv::Scalar>());
+                break;
+            case detail::OpaqueKind::CV_POINT:
+                std::cout << "putOp    cv::Point " << op.args[i].get<cv::Point>().x << " "
+                          << op.args[i].get<cv::Point>().y << " "
+                          << std::endl;
+                sop.opaque_cvpoints.push_back(op.args[i].get<cv::Point>());
+                break;
+            case detail::OpaqueKind::CV_MAT:
+                std::cout << "putOp    cv::Mat " << op.args[i].get<cv::Mat>().rows << " "
+                          << op.args[i].get<cv::Mat>().cols << " "
+                          << op.args[i].get<cv::Mat>().type() << " "
+                          << std::endl;
+                sop.opaque_cvmats.push_back(op.args[i].get<cv::Mat>());
+                break;
+            case detail::OpaqueKind::CV_RECT:
+                std::cout << "putOp    cv::Rect " << op.args[i].get<cv::Rect>().x << " "
+                          << op.args[i].get<cv::Rect>().y << " "
+                          << op.args[i].get<cv::Rect>().width << " "
+                          << op.args[i].get<cv::Rect>().height << " "
+                          << std::endl;
+                sop.opaque_cvrects.push_back(op.args[i].get<cv::Rect>());
+                break;
+            default:
+                std::cout << "putOp    OpaqueKind::UNSUPPORTED" << std::endl;
+            }
+        }
+        else
+        {
+            util::throw_error(std::logic_error("Unexpected ArgKind: expected GOBJREF or OPAQUE"));
+        }
+    }
+
+    for (size_t i = 0; i < op.outs.size(); i++)
+    {
+        const auto rc = op.outs[i];
+        RcDesc src{rc.shape, rc.id};
+        sop.outs[i] = src;
+    }
+
+    for (const auto &in_nh : nh->inNodes())
+    {
+        putData(s, cg, in_nh);
+    }
+
+    for (const auto &out_nh : nh->outNodes())
+    {
+        putData(s, cg, out_nh);
+    }
+
+    s.m_ops.push_back(sop);
+}
+
+void printOp(const Op& op)
```

Comment:
please add the output stream as an `ostream&` parameter and don't refer to `std::cout` directly